File: app/src/main/java/com/lumiyaviewer/lumiya/utils/LevensteinDistance.java
package com.lumiyaviewer.lumiya.utils;

public class LevensteinDistance {
   public static int computeLevensteinDistance(CharSequence var0, CharSequence var1) {
      int[][] var8 = new int[var0.length() + 1][var1.length() + 1];
      int var2 = 0;

      while (var2 <= var0.length()) {
         var8[var2][0] = var2++;
      }

      var2 = 1;

      while (var2 <= var1.length()) {
         var8[0][var2] = var2++;
      }

      for (int var11 = 1; var11 <= var0.length(); var11++) {
         for (int var3 = 1; var3 <= var1.length(); var3++) {
            int[] var9 = var8[var11];
            int var7 = var8[var11 - 1][var3];
            int var5 = var8[var11][var3 - 1];
            int var6 = var8[var11 - 1][var3 - 1];
            byte var4;
            if (var0.charAt(var11 - 1) == var1.charAt(var3 - 1)) {
               var4 = 0;
            } else {
               var4 = 1;
            }

            var9[var3] = minimum(var7 + 1, var5 + 1, var4 + var6);
         }
      }

      return var8[var0.length()][var1.length()];
   }

   private static int minimum(int var0, int var1, int var2) {
      return Math.min(Math.min(var0, var1), var2);
   }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/utils/LevensteinDistance.kt
package com.lumiyaviewer.lumiya.utils

import kotlin.math.min

object LevensteinDistance {
    @JvmStatic
    fun computeLevensteinDistance(charSequence: CharSequence, charSequence2: CharSequence): Int {
        val intsList = Array(charSequence.length + 1) { IntArray(charSequence2.length + 1) }
        var i = 0
        while (i <= charSequence.length) {
            intsList[i][0] = i
            i++
        }
        var j = 1
        while (j <= charSequence2.length) {
            intsList[0][j] = j
            j++
        }
        for (var11 in 1..charSequence.length) {
            for (var3 in 1..charSequence2.length) {
                val ints = intsList[var11]
                val var7 = intsList[var11 - 1][var3]
                val var5 = intsList[var11][var3 - 1]
                val var6 = intsList[var11 - 1][var3 - 1]
                val var4 = if (charSequence[var11 - 1] == charSequence2[var3 - 1]) 0 else 1
                ints[var3] = minimum(var7 + 1, var5 + 1, var4 + var6)
            }
        }
        return intsList[charSequence.length][charSequence2.length]
    }

    private fun minimum(var0: Int, var1: Int, var2: Int): Int {
        return min(min(var0, var1), var2)
    }
}
